ckanext/data_qld/plugin.py

```diff
@@ -0,0 +1,279 @@
+            resource_freshness_helpers.process_next_update_due(data_dict)
+        return search_results
+
+    def delete(self, data_dict):
+        dataset_deletion_helpers.add_deletion_of_dataset_reason(data_dict)
+
+    # IResourceController
+    def before_create(self, context, data_dict):
+        self._check_file_upload(data_dict)
+
+    def before_update(self, context, current_resource, updated_resource):
+        self._check_file_upload(updated_resource)
+        resource_freshness_helpers.check_resource_data(current_resource, updated_resource, context)
+
+    def before_show(self, resource_dict):
+        resource_freshness_helpers.process_nature_of_change(resource_dict)
+        # CKAN background jobs that call 'package_show` will not have request objects
+        if hasattr(request, 'params') and toolkit.get_endpoint()[1] == 'action' and 'package_show' not in request.path:
+            resource_visibility_helpers.process_resource_visibility(resource_dict)
+
+    def _check_file_upload(self, data_dict):
+        # This method is to fix a bug that the ckanext-scheming creates for setting the file size of an uploaded
+        # resource. Currently the actions resource_create and resource_update will only set the resource size if the
+        # key does not exist in the data_dict.
+        # So we will check if the resource is a file upload and remove the 'size' dictionary item from the data_dict.
+        # The action resource_create and resource_update will then set the data_dict['size'] = upload.filesize if
+        # 'size' not in data_dict.
+        file_upload = data_dict.get(u'upload', None)
+        if isinstance(file_upload, uploader.ALLOWED_UPLOAD_TYPES):
+            data_dict.pop(u'size', None)
+
+    # IRoutes
+    def before_map(self, m):
+        # Re_Open a Data Request
+        controller = 'ckanext.data_qld.controller:DataQldUI'
+        m.connect(
+            '/%s/open/{id}' % constants.DATAREQUESTS_MAIN_PATH, controller=controller,
+            action='open_datarequest', conditions=dict(method=['GET', 'POST']))
+
+        m.connect(
+            '/dataset/{dataset_id}/resource/{resource_id}/%s/show/' % constants.SCHEMA_MAIN_PATH,
+            controller=controller, action='show_schema', conditions=dict(method=['GET']))
+
+        # Reporting
+        controller = 'ckanext.data_qld.reporting.controller:ReportingController'
+        m.connect('/dashboard/reporting/export', controller=controller, action='export')
+        m.connect('dashboard.reports', '/dashboard/reporting', controller=controller, action='index')
+        m.connect(
+            '/dashboard/reporting/datasets/{org_id}/{metric}', controller=controller, action='datasets')
+        m.connect(
+            '/dashboard/reporting/datarequests/{org_id}/{metric}', controller=controller, action='datarequests')
+
+        return m
+
+    # IQA
+    def custom_resource_score(self, resource, resource_score):
```

Comment:
can't this custom_resource_score be in its own file like how before_show is called etc

Reply:
It could, but there is currently no associated submodule.

Reply:
was just meaning creating a new class for the function code to live in, so the plugin file is nice an clean.